File: typescript/base/src/main/vsdk/toolkit/common/dataStructures/KDQuery.ts
export class KDQuery {
  public point: number[];
  public wantedN: number;
  public foundN: number;
  public notFilled: boolean;
  public results: unknown[];
  public distances: number[];
  public maximumDistance: number;
  public sqrRadius: number;
  public excludeFlags: number;

  public constructor() {
    this.point = [];
    this.wantedN = 0;
    this.foundN = 0;
    this.notFilled = true;
    this.results = [];
    this.distances = [];
    this.maximumDistance = 0.0;
    this.sqrRadius = 0.0;
    this.excludeFlags = 0;
  }

  public print(): void {
    process.stdout.write(`Point X ${this.point[0]}, Y ${this.point[1]}, Z ${this.point[2]}\n`);
    process.stdout.write(`Wanted N: ${this.wantedN}, found N: ${this.foundN}\n`);
    process.stdout.write(`maximumDistance ${this.maximumDistance}\n`);
    process.stdout.write(`sqrRadius ${this.sqrRadius}\n`);
    process.stdout.write(`excludeFlags ${this.excludeFlags.toString(16)}\n`);
  }
}

File: turboc/base/src/common/dataStructures/KDQuery.cpp
#include "java/lang/System.h"
#include "common/dataStructures/KDQuery.h"

KDQuery::KDQuery():
    point(),
    wantedN(),
    foundN(),
    notFilled(),
    results(),
    distances(),
    maximumDistance(),
    sqrRadius(),
    excludeFlags()
{
}

void
KDQuery::print() const {
    System::out.printf("Point X %g, Y %g, Z %g\n", point[0], point[1], point[2]);
    System::out.printf("Wanted N: %i, found N: %i\n", wantedN, foundN);
    System::out.printf("maximumDistance %g\n", maximumDistance);
    System::out.printf("sqrRadius %g\n", sqrRadius);
    System::out.printf("excludeFlags %x\n", ((int)(excludeFlags)));
}
